### PR TITLE
stm32h743 : add more tests on the nucleo board

### PR DESCRIPTION
**Pull request by FRASTM**
this PR makes the test case with hardware connection pass on the nucleo_h743zi target board from STMicrolectronics
-  tests/drivers/uart/uart_async_api --> connect USART2 D53 / TX (pd5) to D52 / RX (pd6) on the ARDuino connector CN9 of the nucleo 144 board

Each testcase includes the configuration for this board.
Build for the target :
```
west build -p auto -b nucleo_h743zi tests/drivers/uart/uart_async_api/
```
Signed-off-by: Francois Ramu francois.ramu@st.com

### CODE REVIEW

File: tests/drivers/spi/spi_loopback/boards/nucleo_h743zi.conf [hunk 1]
```diff
@@ -0,0 +1,10 @@
+CONFIG_SPI_LOOPBACK_DRV_NAME="SPI_1"
+
+CONFIG_SPI_STM32_DMA=n
```
Comment (erwango) [inline tests/drivers/spi/spi_loopback/boards/nucleo_h743zi.conf:3]: This should not be required as DMA is default n.

File: tests/drivers/spi/spi_loopback/boards/nucleo_h743zi.conf [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+CONFIG_SPI_STM32_INTERRUPT=y
+
+CONFIG_SPI_ASYNC=y
+
+CONFIG_SPI_LOOPBACK_MODE_LOOP=n
+
+CONFIG_NOCACHE_MEMORY=y
```
Comment (erwango) [inline tests/drivers/spi/spi_loopback/boards/nucleo_h743zi.conf:10]: Is this actually required if DMA is not used?

**Review by erwango — CHANGES_REQUESTED** [review]
2 minors comments

File: tests/drivers/spi/spi_loopback/boards/nucleo_h743zi.conf (hunk 1 above)
Comment (FRASTM) [reply tests/drivers/spi/spi_loopback/boards/nucleo_h743zi.conf]: correct

File: tests/drivers/spi/spi_loopback/boards/nucleo_h743zi.conf (hunk 2 above)
Comment (FRASTM) [reply tests/drivers/spi/spi_loopback/boards/nucleo_h743zi.conf]: yes, not needed here

**Review by erwango — APPROVED** [review]

**Comment by FRASTM** [comment]
Add the testcase tests/drivers/spi/spi_loopback  with **DMA** for the stm32h743zi nucleo board :
`west build -p auto -b nucleo_h743zi tests/drivers/spi/spi_loopback/`

--> connect pin (D11) SPI 1 MOSI to (D12)  SPI1 MISO on the target board to pass the test.

**Review by ABOSTM — APPROVED** [review]

**Comment by erwango** [comment]
@FRASTM Can you rebase ?

**Comment by FRASTM** [comment]
rebased on 479215100d132044889750e88d7a48f7b4913232
--> only  tests/drivers/uart/uart_async_api  
(the spi loopback for the stm32h743 is now in the PR  https://github.com/zephyrproject-rtos/zephyr/pull/41569)

**Pull request merged**